src/util/syntaxhelp.h

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2022 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_UTIL_SYNTAXHELP_H
+#define BITCOIN_UTIL_SYNTAXHELP_H
+
+/* Syntactic sugar / helpers
+ * Perhaps can be replaced by C++20 designated initializers.
+ */
+
+namespace util {
+
+/* Fakes named arguments in C++.
+ *
+ * If you have:
+ *
+ *    bool _f(int x, int y) { ... }
+ *
+ * Then writing:
+ *
+ *    static constrexpr NamedArgument<int, struct x_arg> _x;
+ *    static constrexpr NamedArgument<int, struct y_arg> _y;
+ *    bool f(decltype(_x)::Arg x, decltype(_y)::Arg y) { return _f(x.v, y.v); }
+ *
+ * allows you to invoke f(_x=3, _y=4) instead of _f(3, 4) and
+ * will give a compile time error if you reverse the labels
+ * (ie, f(_y=3, _x=4)).
+ */
+
+template<typename T, typename TAG>
+struct NamedArgument
+{
+    struct Arg {
+        using Tag = TAG;
+        const T v;
+        Arg(T&& v) : v{std::move(v)} { }
+        Arg(const T& v) : v{v} { }
+    };
+
+    Arg operator=(T&& v) const
+    {
+        return Arg(std::move(v));
+    }
+    Arg operator=(const T& v) const
+    {
+        return Arg(v);
+    }
+};
+
+/* Helper for std::array initialization.
+ *
+ * Allows for specifying the expected index of each initializer to make
+ * review easier, and will give a compile-time error if any initializers
+ * are missing, extraneous or out of order.
+ */
+
+template<size_t O, typename T>
+struct ArrayInitElement
+{
+    static constexpr size_t offset = O;
+    const T t;
+
+    template<typename... I>
+    ArrayInitElement(I... i) : t{i...} { };
+};
+
+template<typename T, size_t N>
+class ArrayInitHelper
+{
+private:
+    using Arr = std::array<T, N>;
+
+    template<size_t E>
+    static inline void step(Arr& a)
+    {
+        static_assert(E == N, "too few initializers for array");
+    }
+
+    template<size_t E, size_t O, typename... I>
+    static inline void step(Arr& a, const ArrayInitElement<O, T>& aie, const I&... i)
+    {
+        static_assert(E < N, "too many initializers for array");
+        static_assert(E == O, "array initializer element is out of order");
+        if constexpr (E < N) {
```

Comment:
FYI: this `if constexpr` is only here to avoid redundant warnings that `std::get<N>` and `step<N+1>` would generate.